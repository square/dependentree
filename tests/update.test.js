const puppeteer = require('puppeteer');

const pageURL = 'http://127.0.0.1:8081/index.html';


// Not much to test with update. Just ensure clicking on nodes
// triggers update which changes the tree
// Note that nodes that are not visible will still show up
// in the snapshots
describe('Update', () => {
  let one;
  let two;
  let three;

  beforeAll(async () => {
    const browser = await puppeteer.launch({ args: ['--no-sandbox'] });
    const page = await browser.newPage();
    await page.goto(pageURL);
    ({ one, two, three } = await page.evaluate(() => {
      const tree = new DependenTree('div#tree');
      tree.addEntities(royals, { animationDuration: 0 });
      tree.setTree('Elizabeth II', 'downstream');

      const one = tree.svg.node().innerHTML;

      // clicking on Charles
      tree.svg.select('g:nth-of-type(3) > circle').dispatch('click');
      const two = tree.svg.node().innerHTML;

      // clicking on Elizabeth II
      tree.svg.select('circle').dispatch('click');
      const three = tree.svg.node().innerHTML;

      return { one, two, three };
    }));

    await page.close();
    await browser.close();
  });

  test('default tree', () => {
    expect(one).toMatchSnapshot();
  });

  test('open node', () => {
    expect(two).toMatchSnapshot();
  });

  test('close root', () => {
    expect(three).toMatchSnapshot();
  });
});
