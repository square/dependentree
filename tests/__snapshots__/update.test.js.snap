// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update close root 1`] = `
"<path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 180 -45
    C 180 -45, 180 -45, 180 -45
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 180 -45
    C 180 -45, 180 -45, 180 -45
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><g></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Elizabeth II</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: white;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Charles</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Anne</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Andrew</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Edward</tspan></text></g><g class=\\"node\\" transform=\\"translate(180,-45)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">William</tspan></text></g><g class=\\"node\\" transform=\\"translate(180,-45)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Harry</tspan></text></g>"
`;

exports[`Update default tree 1`] = `
"<path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><g></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: white;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Elizabeth II</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Charles</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Anne</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Andrew</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Edward</tspan></text></g>"
`;

exports[`Update open node 1`] = `
"<path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 0 0
    C 0 0, 0 0, 0 0
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 180 -45
    C 180 -45, 180 -45, 180 -45
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><path d=\\"
    M 180 -45
    C 180 -45, 180 -45, 180 -45
  \\" style=\\"fill: none; stroke: rgb(221, 221, 221); stroke-width: 2px;\\"></path><g></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: white;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Elizabeth II</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: white;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Charles</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Anne</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Andrew</tspan></text></g><g class=\\"node\\" transform=\\"translate(0,0)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Edward</tspan></text></g><g class=\\"node\\" transform=\\"translate(180,-45)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">William</tspan></text></g><g class=\\"node\\" transform=\\"translate(180,-45)\\" cursor=\\"pointer\\"><circle r=\\"10\\" style=\\"stroke: steelblue; stroke-width: 3px; fill: lightsteelblue;\\"></circle><text dy=\\".35em\\" x=\\"-13\\" text-anchor=\\"end\\" fill=\\"black\\" style=\\"fill-opacity: 1; font: 12px sans-serif;\\"><tspan x=\\"-13\\">Harry</tspan></text></g>"
`;
